### PR TITLE
fix(redis-controller): fix npe issue and some logic error

### PR DESCRIPTION
**Pull request by chlins**
Signed-off-by: chlins <chlins.zhang@gmail.com>

Fix: https://github.com/goharbor/harbor-operator/issues/204

### CODE REVIEW

**Review by Ghostbaby — APPROVED** [review]
lgtm

File: pkg/cluster/controllers/cache/redis.go [hunk 1]
```diff
@@ -47,13 +47,14 @@ type RedisController struct {
 	Scheme             *runtime.Scheme
 	RedisConnect       *RedisConnect
 	ResourceManager    ResourceManager
-	expectCR, actualCR *redisOp.RedisFailover
+	expectCR, actualCR runtime.Object
```
Comment (steven-zou) [inline pkg/cluster/controllers/cache/redis.go:50]: Why use `runtime.Object` instead of `redisOp.RedisFailover`?

**Review by steven-zou — COMMENTED** [review]
one comment

File: pkg/cluster/controllers/cache/redis.go (hunk 1 above)
Comment (chlins) [reply pkg/cluster/controllers/cache/redis.go:50]: If use redisOp.RedisFailover, when controller gets actual cr from crd client, the type of crd client return if unstructed.Unstructed, which implements runtime.Object, so we can directly use rc.actualCR=${crdReturn}.

Comment (steven-zou) [reply pkg/cluster/controllers/cache/redis.go:50]: 👌

**Review by steven-zou — APPROVED** [review]
LGTM

**Pull request merged**